src/cycle/osd-scroll.c

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <wlr/types/wlr_scene.h>
+#include "common/lab-scene-rect.h"
+#include "labwc.h"
+#include "cycle.h"
+#include "output.h"
+
+void
+cycle_osd_scroll_init(struct output *output, struct wlr_box bar_area,
+		int delta_y, int nr_cols, int nr_rows, int nr_visible_rows,
+		float *border_color, float *bg_color)
+{
+	if (nr_visible_rows >= nr_rows) {
+		/* OSD doesn't have so many windows to scroll through */
+		return;
+	}
+
+	struct cycle_osd_scroll_context *scroll = &output->cycle_osd.scroll;
+	scroll->nr_cols = nr_cols;
+	scroll->nr_rows = nr_rows;
+	scroll->nr_visible_rows = nr_visible_rows;
+	scroll->top_row_idx = 0;
+	scroll->bar_area = bar_area;
+	scroll->delta_y = delta_y;
+	scroll->bar_tree = wlr_scene_tree_create(output->cycle_osd.tree);
+	wlr_scene_node_set_position(&scroll->bar_tree->node,
+		bar_area.x, bar_area.y);
+
+	struct lab_scene_rect_options scrollbar_opts = {
+		.border_colors = (float *[1]) { border_color },
```

Comment:
Could we pass `float **border_color` to `cycle_osd_scroll_init()` and then simply do `.border_colors = border_color` here.

I might be missing something about how the code works here.